components/vtpm2-server/run-jss-server.sh

```diff
@@ -0,0 +1,59 @@
+#!/bin/sh
+# this script checks if X5c file exists.
+# if YES, the Private server starts and Public server shuts down
+# if NO, the Public server starts and Private server shuts down
+REPEAT_SECS=30
+STATEDIR="${STATEDIR:-/tmp}"
+X5CFILE="${STATEDIR}/x5c"
+
+start_pub_server() {
+  if ps -ef | grep flask | grep -v grep; then
+    echo "web server already running"
+  else
+    cd /usr/local/bin || exit
+    FLASK_APP=/jss-server-pub.py python -m flask run --host=0.0.0.0 --port=5000 &
+  fi
+}
+
+start_priv_server() {
+  if ps -ef | grep uwsgi | grep -v grep; then
+    echo "socket server already running"
+  else
+    cd /usr/local/bin || exit
+    uwsgi --http-socket /host/sockets/app.sock --chmod-socket=666 --manage-script-name --mount /=jss-server-priv:app --plugins python &
+  fi
+}
+
+stop_pub_server() {
+  if ps -ef | grep flask | grep -v grep; then
+    echo "stopping the running web server..."
+    kill -9 $(ps -ef |grep flask | grep -v grep | awk '{print $2}')
+  else
+    echo "web server not running"
+  fi
+}
+
+stop_priv_server() {
+  if ps -ef | grep uwsgi | grep -v grep; then
+    echo "stopping the running socket server..."
+    kill -9 $(ps -ef |grep uwsgi | grep -v grep | awk '{print $2}')
```

Comment:
I'd prefer if we stored PIDs, but this is fine for now. Let's put this in a variable as a command at the start of file for better readability like:
```
cmd_get_priv_server_pid="ps -ef |grep uwsgi | grep -v grep | awk '{print $2}'"
cmd_get_pub_server_pid="ps -ef |grep uwsgi | grep -v grep | awk '{print $2}'"
```

and then we can use these as well in the if statements to be like
`if $cmd_get_pub_server_pid is empty` or something so we end up having only 2 commands to use to find the uswgi and flask servers throughout the code?

Reply:
Done